Applications/MIDIMonitor/SMMAppController.m

```diff
@@ -357,4 +368,15 @@ - (void)openWindowForNewlyAppearedSources
     self.newlyAppearedSources = nil;
 }
 
+- (void)autoConnectToNewlyAppearedSources
+{
+    NSDocumentController *dc = [NSDocumentController sharedDocumentController];
+    SMMDocument *document = [dc currentDocument];
+    [document setSelectedInputSources:self.newlyAppearedSources];
+    SMMMonitorWindowController *wc = document.windowControllers.firstObject;
```

Comment:
If there are multiple windows, you are doing this for only one of them. Shouldn't it change all of them?

Reply:
My thought was to autoconnect to new devices in the current window only. But I can change it so that it works with all windows.

Reply:
OK, current window is probably fine; just wanted to make sure that whatever you were doing, you were doing it intentionally. 

It's possible that `[dc currentDocument]` is nil (for instance if no windows are open) so make sure the code does something sensible if that's the case.

Reply:
Good point. Nothing happens if there isn't a currently selected window. I'll see about making it select the most recently used window. What do you think?

Reply:
Sounds good!  Whatever seems like the most reasonable user experience.